src/robusta/utils/scope.py

```diff
@@ -59,12 +54,12 @@ def scope_matches(self, scope: ScopeIncludeExcludeParamsT) -> bool:
     def scope_attribute_matches(self, attr_name: str, attr_matchers: List[str]) -> bool:
         data = self.get_data()
         if attr_name not in data:
-            logging.warning(f'Scope match on non-existent attribute "{attr_name}" ({data=}')
+            logging.warning(f'Scope match on non-existent attribute "{attr_name}" ({data=})')
             return False
         attr_value = data[attr_name]
         for attr_matcher in attr_matchers:
-            if attr_name == "attributes":
-                return self.scope_match_attributes(attr_matcher, attr_value)
+            if hasattr(self, f"match_{attr_name}"):
```

Comment:
please leave it more explicit

I think it's best to load the namespace labels only here, of there's actually a matcher that uses it
It will be used on a fraction of the users only

Reply:
That way we would be accumulating abstract methods in the base class, each of which is implemented in some random derived class. Are you sure we want to go that way? The mechanism reintroduced here, `match_{whatever}` IMO elegantly implements the matching of attributes that is not explicitly implemented in the base class, basing on attribute name.

Reply:
Okay, for reasons of clarity changed the `scope_match_*` methods/calls back to be explicit.